### PR TITLE
On branch edburns-msft-gh-618-jdbc-version

### PR DESCRIPTION
**Pull request by edburns**
modified:   specification/pom.xml
- Increment versions.
modified:   specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc
- Remove reference to JDBC 4.2.
- Replace with verbiage asking the reader to look up the version from the package javadoc for `java.sql`, which does state the version and provides a link to the corresponding JSR.

### CODE REVIEW

**Review by jhanders34 — APPROVED** [review]

File: specification/pom.xml [hunk 1]
```diff
@@ -20,14 +20,14 @@
...
-        <version>1.0.6</version>
+        <version>1.0.7</version>
         <relativePath/>
     </parent>
     <modelVersion>4.0.0</modelVersion>
     <groupId>jakarta.platform</groupId>
     <artifactId>platform-specs</artifactId>
     <packaging>pom</packaging>
-    <version>11.0</version>
+    <version>11.0.0-M1</version>
```
Comment (Emily-Jiang) [inline specification/pom.xml:30]: Should the version be 11.0-M1?

**Comment by edburns** [comment]
Hello @Emily-Jiang , thanks for catching and clarifying. Please review so I can merge this.

Thanks,

Ed

**Review by Emily-Jiang — APPROVED** [review]
LGTM

File: specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc [hunk 2]
```diff
@@ -981,8 +981,7 @@ Connector architecture must support deploying and using a JDBC driver
 that has been written and packaged as a resource adapter using the
 Connector architecture.
 
-The JDBC 4.2 specification is available at
-_https://jcp.org/en/jsr/detail?id=221_ .
+Every release of Jakarta EE declares a minimum required version of Java SE. For discussion, let this be Java SE N. Compatible implementations of Jakarta EE must support the latest version of the JDBC API mentioned in the Java SE N javadocs for the package `java.sql`. These javadocs typically have a link to the corresponding specification at `jcp.org`.
```
Comment (hs536) [inline specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc:984]: @edburns Hi, does this statement mean that, for example, if the following javadoc is given, compliance with the latest JDBC 4.3 spec becomes a condition for the approval of the platform implementation?
https://docs.oracle.com/en/java/javase/21/docs/api/java.sql/java/sql/package-summary.html

**Pull request merged**